probe-rs/src/config/flash_algorithm.rs

```diff
@@ -0,0 +1,34 @@
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct FlashAlgorithm {
+    /// Memory address where the flash algo instructions will be loaded to.
+    pub load_address: u32,
+    /// List of 32-bit words containing the position-independant code for the algo.
+    pub instructions: Vec<u32>,
+    /// Address of the `Init()` entry point. Optional.
+    pub pc_init: Option<u32>,
+    /// Address of the `UnInit()` entry point. Optional.
+    pub pc_uninit: Option<u32>,
+    /// Address of the `ProgramPage()` entry point.
+    pub pc_program_page: u32,
+    /// Address of the `EraseSector()` entry point.
+    pub pc_erase_sector: u32,
+    /// Address of the `EraseAll()` entry point. Optional.
+    pub pc_erase_all: Option<u32>,
+    /// Initial value of the R9 register for calling flash algo entry points, which
+    /// determines where the position-independant data resides.
+    pub static_base: u32,
+    /// Initial value of the stack pointer when calling any flash algo API.
+    pub begin_stack: u32,
+    /// Base address of the page buffer. Used if `page_buffers` is not provided.
```

Comment:
This also has a min. size of `page_size`, right?

Reply:
Do you mean the `begin_data` address? Not sure if GH is tricking me. It is ensured during flash algo parsing that it has at least `page_size` bytes space for the page. Maybe we could move this elsewhere.

Reply:
I fixed this as well with the completely position independant flash algo.